up-streamer/src/ustreamer.rs

```diff
@@ -209,12 +207,13 @@ impl ForwardingListeners {
         let forwarding_listener =
             Arc::new(ForwardingListener::new(forwarding_id, out_sender.clone()));
 
-        type UUriPair = Vec<(UUri, Option<UUri>)>;
-        let mut uuri_to_backpedal: HashSet<UUriPair> = HashSet::new();
+        type SourceSinkFilterPair = (UUri, Option<UUri>);
+        #[allow(clippy::mutable_key_type)]
+        let mut uuris_to_backpedal: HashSet<SourceSinkFilterPair> = HashSet::new();
```

Comment:
Do you have some unit test / integration test to prove this mechanism works?

Reply:
just added one